lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -897,6 +897,22 @@ def __init__(self, charm: CharmBase, relationship_name: str):
         self.charm = charm
         self.relationship_name = relationship_name
 
+    def _load_relation_data(self, raw_relation_data: dict) -> dict:
```

Comment:
There might be cases when `_load_relation_data` should work for non-leader-units.
For example a non-leader unit trying to read the unit databag of the same application or reading from the remote application's databags.
Please refer to [this diagram ](https://discourse.charmhub.io/uploads/default/original/2X/c/cc50a23c3d5ce4458dd0ca81f3d3ff1768ad0607.png) from the [docs](https://juju.is/docs/sdk/integration).

I would suggest adding guards where we try to call this function with `some_relation.data[self.charm.app]`

Reply:
Since it is a private method, I am not sure we need to support those cases, as there does not seem to be any of those cases in this library.

Do you have an example in mind?

Reply:
Yes I do.

there is this line in the `request_certificate` of the requirer
`relation_data = _load_relation_data(relation.data[self.model.unit])`
I don't see why `request_certificate`  would be only done by leader units.

But why should `_load_relation_data` know about the unit and if it's leader or not? in my opinion we should guard the calls of `_load_relation_data`.

Reply:
@saltiyazan Thanks for indication the problem here. I agree, we need to add guard only  if we call some_relation.data[self.charm.app]. 
The PR is updated accordingly.